docs/develop/smart-contracts/security/things-to-focus.md

```diff
@@ -0,0 +1,76 @@
+# Things to Focus on while Working with TON Blockchain
+
+In this article, we will review and discuss the elements to consider for those who want to develop TON applications.
+
+## Checklist
+
+### 1. Name collisions
+
+Func variables and functions may contain almost any legit character. I.e. `var++`, `~bits`, `foo-bar+baz` including commas are valid variables and functions names.
+
+When writing and inspecting a Func code, Linter should be used
+
+### 2. Check the throw values
+
+Each time the TVM execution stops normally, it stops with exit codes `0` or `1`. Although it is done automatically, TVM execution can be interrupted directly in an unexpected way if exit codes `0` and `1` are thrown directly by either `throw(0)` or `throw(1)` command.
+
+### 3. Func is a strictly typed language with data structures holding exactly what they are supposed to store
+
+It is crucial to keep track of what the code does and what it may return. Keep in mind that the compiler cares only about the code and only in its initial state. After certain operations stored values of some variables can change.
+
+Reading unexpected variables' values and calling methods on data types that are not supposed to have such methods (or their return values are not stored properly) are errors and are not skipped as "warnings" or "notices" but lead to unreachable code. Keep in mind that storing an unexpected value may be okay, however, reading it may cause problems e.g. error code 5 (integer out of expected range) may be thrown for an integer variable.
+
+### 4. Unlike other blockchains, TON does not contain revert messages, only exit codes
```

Comment:
I believe this point should go after 6,7 and one more:

### 6-1. TON contracts are autonomous

Contracts in the blockchain can reside in separate shards, processed by other set of validators, meaning that developer cannot pull data from other contracts on demand. Thus, any communication is asynchronous and done by sending messages.